e-library/src/Components/homepage/body.jsx

```diff
@@ -1,58 +1,46 @@
-import React from "react";
-import Button from "../button/button";
-import Images from '../images/images'
+import React, { useEffect, useState } from "react";
+import axios from "../api/axios";
+
+export const userInfo = JSON.parse(localStorage.getItem('user-info'))
+
 const Body = () => {
+    const [book, setBook] = useState([]);
+
+        useEffect(() => {
+            const getBooks = async () => {
+                try{
+                const response = await axios.get('api/books', {
+                    headers: {
+                        'authorization': userInfo.token,
+                    }
+                })
+                setBook(response.data.books)
+            } catch (err) {
+                console.log(err)
+            }
+        }
+        getBooks();
+        }, [])
```

Comment:
move to a custom hook. integrate Redux into the `fetch` process. the data should be saved in the redux state not react's state.